scipy-stubs/fft/_realtransforms.pyi

```diff
@@ -1,82 +1,84 @@
-from scipy._typing import Untyped
+from numpy import float64, generic
+from numpy.typing import NDArray
+from scipy._typing import DCTType, NormalizationMode, Untyped
 
 def dctn(
     x: Untyped,
-    type: int = 2,
+    type: DCTType = 2,
     s: Untyped | None = None,
     axes: Untyped | None = None,
-    norm: Untyped | None = None,
+    norm: NormalizationMode | None = None,
     overwrite_x: bool = False,
     workers: Untyped | None = None,
     *,
     orthogonalize: Untyped | None = None,
 ) -> Untyped: ...
 def idctn(
     x: Untyped,
-    type: int = 2,
+    type: DCTType = 2,
     s: Untyped | None = None,
     axes: Untyped | None = None,
-    norm: Untyped | None = None,
+    norm: NormalizationMode | None = None,
     overwrite_x: bool = False,
     workers: Untyped | None = None,
     orthogonalize: Untyped | None = None,
 ) -> Untyped: ...
 def dstn(
     x: Untyped,
-    type: int = 2,
+    type: DCTType = 2,
     s: Untyped | None = None,
     axes: Untyped | None = None,
-    norm: Untyped | None = None,
+    norm: NormalizationMode | None = None,
     overwrite_x: bool = False,
     workers: Untyped | None = None,
     orthogonalize: Untyped | None = None,
 ) -> Untyped: ...
 def idstn(
     x: Untyped,
-    type: int = 2,
+    type: DCTType = 2,
     s: Untyped | None = None,
     axes: Untyped | None = None,
-    norm: Untyped | None = None,
+    norm: NormalizationMode | None = None,
     overwrite_x: bool = False,
     workers: Untyped | None = None,
     orthogonalize: Untyped | None = None,
 ) -> Untyped: ...
 def dct(
-    x: Untyped,
-    type: int = 2,
-    n: Untyped | None = None,
+    x: NDArray[generic],
```

Comment:
the [docs](https://docs.scipy.org/doc/scipy/reference/generated/scipy.fft.dct.html) describe `x` as an "`array_like`". So that means that e.g. `[1, 2, 3]` is also allowed here. 

```suggestion
    x: npt.ArrayLike,
```

If you want to, you could narrow this down even further by requiring it to be numeric (i.e. integral, real, or complex), by using the `_ArrayLikeNumber_co` alias from `numpy._typing` instead (don't worry about it being private).